x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java

```diff
@@ -272,8 +272,11 @@ static IndexResolution mergedMapping(String indexPattern, Map<String, Map<String
 
             String name = entry.getKey();
 
-            // skip internal fields
-            if (!name.startsWith("_")) {
+            // Skip internal fields (name starting with underscore and its type reported by field_caps starts with underscore
+            // as well). A meta field named "_version", for example, has the type named "_version".
+            // Also, skip any of the blacklisted field names.
+            if (false == (name.startsWith("_") && entry.getValue().values().stream().anyMatch(f -> f.getType().startsWith("_"))) &&
```

Comment:
Potentially move the `types` declaration above the if to reduce the if:
`types.values().stream().anyMatch()`

Another option would be to do the type check further down when it is encountered - this would avoid creating a stream for _every_ single field.

Is the name check still necessary ? Wouldn't the type check be enough.

Reply:
Seems better this way.